src/test/components/elements/EventBookingForm.test.tsx

```diff
@@ -175,8 +170,42 @@ describe("EventBookingForm", () => {
 
   it("if event is not virtual, it displays a message to advise PII information will be deleted after 30 days", () => {
     const event = augmentEvent(mockEvent);
-    setupTest({ role: "STUDENT", user: mockUser, event: { ...event, isVirtual: false } });
+    setupTest({ role: "STUDENT", user: mockUserSummary, event: { ...event, isVirtual: false } });
     const piiMessage = screen.getByTestId("pii-message");
     expect(piiMessage).toBeInTheDocument();
   });
+
+  it("shows an error message if first name is missing", () => {
+    const targetUser = { ...mockUserSummary, givenName: undefined };
+    setupTest({ role: "STUDENT", user: targetUser });
+    const formFeedback = screen.getByText("First name is required");
+    expect(formFeedback).toBeInTheDocument();
```

Comment:
Better to change this (and following tests) to a check on whether the text box is invalid, rather than presence of the error message as that will always pass due to the FormFeedback component of Reactstrap.

Reply:
I tried adding `it.each` tests, however it wants be to set up `beforeEach` tests. which makes sense however if I use that outside of the test, it is being used for everything else too, which is then making them false. 

I am unable to nest the `beforeEach` inside the `it` test as it it is wrong, and wont work.

Only solution I could think of would be to separate the tests out, into individual tests looking at the form feedback with `.toBevalid`

Reply:
You shouldn't need a `beforeEach` here? You'd be looking at something like this 

```
  const missingDetailsTestCases = [
    { field: "First name", user: { ...mockUserSummary, givenName: undefined } },
    { field: "Last name", user: { ...mockUserSummary, familyName: undefined } },
  ];
  ```

And so on with more cases, then: 

```  
  it.each(missingDetailsTestCases)("has an invalid form field if $field is missing", ({ user, field }) => {
    setupTest({ role: "STUDENT", user: user });
    const invalidField = screen.getByRole("textbox", { name: field });
    expect(invalidField).toBeInvalid();
  });

```
  
 You can absolutely do them as individual tests if you prefer, this can just be a little more concise if there's a series of very similar tests.

Reply:
Used that to create the tests for failing and passing.